src/main/java/de/tum/cit/aet/artemis/buildagent/dto/BuildJobsStatisticsDTO.java

```diff
@@ -20,19 +18,21 @@ public static BuildJobsStatisticsDTO of(List<BuildJobResultCountDTO> resultCount
         long successfulBuilds = 0;
         long failedBuilds = 0;
         long cancelledBuilds = 0;
-        // Switch case would cause an error in the testDTOImplementations test
+        long timeOutBuilds = 0;
+        long missingBuilds = 0;
+        long otherBuilds = 0;
+
         for (BuildJobResultCountDTO resultCountDTO : resultCountDTOList) {
-            if (resultCountDTO.status() == BuildStatus.SUCCESSFUL) {
-                successfulBuilds += resultCountDTO.count();
-            }
-            else if (resultCountDTO.status() == BuildStatus.FAILED || resultCountDTO.status() == BuildStatus.ERROR) {
-                failedBuilds += resultCountDTO.count();
-            }
-            else if (resultCountDTO.status() == BuildStatus.CANCELLED) {
-                cancelledBuilds += resultCountDTO.count();
+            switch (resultCountDTO.status()) {
+                case SUCCESSFUL -> successfulBuilds += resultCountDTO.count();
+                case FAILED, ERROR -> failedBuilds += resultCountDTO.count();
+                case CANCELLED -> cancelledBuilds += resultCountDTO.count();
+                case TIMEOUT -> timeOutBuilds += resultCountDTO.count();
+                case MISSING -> missingBuilds += resultCountDTO.count();
+                default -> otherBuilds += resultCountDTO.count();
             }
         }
-        totalBuilds = successfulBuilds + failedBuilds + cancelledBuilds;
-        return new BuildJobsStatisticsDTO(totalBuilds, successfulBuilds, failedBuilds, cancelledBuilds);
+        totalBuilds = successfulBuilds + failedBuilds + cancelledBuilds + timeOutBuilds + missingBuilds + otherBuilds;
+        return new BuildJobsStatisticsDTO(totalBuilds, successfulBuilds, failedBuilds, cancelledBuilds, timeOutBuilds, missingBuilds);
```

Comment:
would it make sense to also send the `otherBuilds` here? those are the ones currently BUILDING and QUEUED if I understand correctly? 
Would it make sense to also count them individually?

Reply:
I dont think It would be necessary since they are shown in the view below (processing builds and queued builds)